solveengine/mipmodel.py

```diff
@@ -8,8 +8,10 @@
 
 from enum import Enum
 from collections import namedtuple
+from numpy import ndarray
```

Comment:
do we really need this dependency? The other way of modelling does not need this. 
(see below)

at the very least it should be in the requirements.txt